utils/test_harness/tools/src/test_harness_debug.cpp

```diff
@@ -141,6 +142,70 @@ void debug_resume(const zet_debug_session_handle_t &debug_session,
   EXPECT_EQ(ZE_RESULT_SUCCESS, zetDebugResume(debug_session, device_thread));
 }
 
+void printRegSetProperties(zet_debug_regset_properties_t regSet) {
+
+  LOG_DEBUG << "[Debugger] Reg Set Type: " << regSet.type;
+  LOG_DEBUG << "[Debugger] Reg Set Version: " << regSet.version;
+  LOG_DEBUG << "[Debugger] Reg Set GeneralFlags: " << regSet.generalFlags;
+  LOG_DEBUG << "[Debugger] Reg Set deviceFlags: " << regSet.deviceFlags;
+  LOG_DEBUG << "[Debugger] Reg Set count: " << regSet.count;
+  LOG_DEBUG << "[Debugger] Reg Set bit size: " << regSet.bitSize;
+  LOG_DEBUG << "[Debugger] Reg Set byte size: " << regSet.byteSize;
+  return;
+}
+
+bool get_register_set_props(ze_device_handle_t device,
+                            zet_debug_regset_type_intel_gpu_t type,
+                            zet_debug_regset_properties_t &reg) {
+  uint32_t nRegSets = 0;
+  zetDebugGetRegisterSetProperties(device, &nRegSets, nullptr);
+  zet_debug_regset_properties_t *pRegSets =
+      new zet_debug_regset_properties_t[nRegSets];
+  for (int i = 0; i < nRegSets; i++) {
+    pRegSets[i] = {ZET_STRUCTURE_TYPE_DEBUG_REGSET_PROPERTIES, nullptr};
+  }
+  zetDebugGetRegisterSetProperties(device, &nRegSets, pRegSets);
+
+  bool found = false;
+  for (int i = 0; i < nRegSets; i++) {
+    if (pRegSets[i].type == type) {
+      printRegSetProperties(pRegSets[i]);
+      reg = pRegSets[i];
+      found = true;
+      break;
+    }
+  }
+
+  delete[] pRegSets;
+  return found;
+}
+
+void clear_exceptions(const ze_device_handle_t &device,
+                    const zet_debug_session_handle_t &debug_session,
+                  const ze_device_thread_t &device_thread) {
```

Comment:
I'm pretty sure device_thread has to be a single thread, so if thread_all ({UINT32_MAX, UINT32_MAX, UINT32_MAX, UINT32_MAX}) is passed in you should iterate over those threads

Reply:
So now only place from where this method could be called is get_stopped_threads , from where we are not passing thread_all?

Reply:
Ok, makes sense now.